packages/core/src/api/plugin/Plugin.tsx

```diff
@@ -34,6 +35,12 @@ export type PluginHooks = {
 };
 
 export type RouterHooks = {
+  addRoute(
```

Comment:
Proposal here is to rename `registerRoute` to `addRoute`, and at the same time switch all registered routes to using nav targets. Instead of overloading `registerRoute` I figured we just do both migrations at once, and remove `registerRoute` once it's no longer used.

Reply:
Plenty of time to backtrack on this after merging, but as mentioned during sync this actually removes the possibility of registering routes using a hardcoded path. This means there needs to be very little friction to using nav target and registering them, and I think the new API needs more work there.